### PR TITLE
Add remaining domintro boxes

### LINKED ISSUE #543: Add domintro explanations for web developers
I think the API would greatly benefit from some domintro boxes. This would allow you to add some developer-facing documentation for the more confusing concepts that end up manifesting in the API, like a response's type. The domintro could actually explain what each value you get back means. (I wasn't able to figure out the difference between "basic" and "default" myself.)

### PR DESCRIPTION
**Pull request by annevk**
For the Headers class, Body mixin, Request's clone(), and the Response class.

Closes #543.

----

I would very much appreciate review in the form of fixup commits as this is extremely tedious work. 😊 I didn't really try to tackle any greater style questions, though I would be (more) inclined to address suggestions there as I realize that could use some renewal, especially now that JavaScript has destructuring.

***
<a href="https://whatpr.org/fetch/1161.html" title="Last updated on Feb 4, 2021, 10:12 AM UTC (a2090d5)">Preview</a> | <a href="https://whatpr.org/fetch/1161/89cfc1d...a2090d5.html" title="Last updated on Feb 4, 2021, 10:12 AM UTC (a2090d5)">Diff</a>

### CODE REVIEW

File: fetch.bs [hunk 1]
```diff
@@ -5279,6 +5263,51 @@ objects.</span>
...
+
+ <dt><code><var>headers</var> . <a method for=Headers lt=append()>append</a>(<var>name</var>, <var>value</var>)</code>
+ <dd><p>Appends a header to <var>headers</var>.
+
+ <dt><code><var>headers</var> . <a method for=Headers lt=delete()>delete</a>(<var>name</var>)</code>
+ <dd><p>Removes a header from <var>headers</var>.
+
+ <dt><code><var>headers</var> . <a method for=Headers lt=get()>get</a>(<var>name</var>)</code>
+ <dd><p>Returns the values of all headers whose name is <var>name</var>, separated by a comma and a
+ space.
```
Comment (ricea) [inline fetch.bs:5296]:
```suggestion
 <dd><p>Returns as a string the values of all headers whose name is <var>name</var>, separated by a
 comma and a space.
```

File: fetch.bs [hunk 2]
```diff
@@ -5279,6 +5263,51 @@ objects.</span>
...
+
+ <dt><code><var>headers</var> . <a method for=Headers lt=has()>has</a>(<var>name</var>)</code>
+ <dd><p>Returns whether there is a header whose name is <var>name</var>.
+
+ <dt><code><var>headers</var> . <a method for=Headers lt=set()>set</a>(<var>name</var>, <var>value</var>)</code>
+ <dd><p>Replaces the value of the first header whose name is <var>name</var> with <var>value</var>
+ and removes any remaining headers whose name is <var>name</var>.
+
+ <dt><code>for([<var>name</var>, <var>value</var>] of <var>headers</var>)</code>
```
Comment (ricea) [inline fetch.bs:5305]:
```suggestion
 <dt><code>for(const [<var>name</var>, <var>value</var>] of <var>headers</var>)</code>
```

**Review by ricea — APPROVED** [review]
lgtm with suggestions.

**Comment by annevk** [comment]
Thanks!

**Pull request merged**